h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -146,6 +146,23 @@ public static boolean haveMojo(String urlName) {
     return BUILDERS[ensureBuilderIndex(urlName)].haveMojo();
   }
 
+  /**
+   * Helper method that enables preprocessing with TE model. It's user's responsibility to make sure TE is trained in an appropriate way to avoid data leakage.
+   * E.g. TE without KFold strategy being applied for cv main model leads to a data leakage.
+   * @param model key to the trained TE model
+   */
+  public void addTEModelKey(Key<Model> model) {
+    _parms._te_model = model;
+  }
+
+  /**
+   *
+   * @return TE model's key if one was assigned to a model builder, null otherwise
+   */
+  public Key<Model> getTEModelKey() {
+    return _parms._te_model;
+  }
+
```

Comment:
as this is supposed to be temporary, can we make those methods "less public" as suggested by @michalkurka ?
same with `deleteTEModel` in `Model` class and the method in `FrameUtils`.
Suggestion is to use naming convention: prefix methods with "_" or "internal".

Reply:
@sebhrusen  I thought we were talking about Py and R API's. But probably you are right and it extends to Java API as well. It's just the fact that we don't provide docs for Java api makes me feel that Java api is not for customers/users. Thanks, will change naming.

Reply:
totally agree, but still safer with an explicit naming convention.

Reply:
and some few customers use the Java API directly, without mentioning those who extend/fork the project (see internal projects at H2O).